packages/account-abstraction/src/BaseAccountAPI.ts

```diff
@@ -258,12 +258,20 @@ export abstract class BaseAccountAPI {
     if (!detailsForUserOp.gasLimit) {
       // TODO : error handling
       // Capture the failure and throw message
-      callGasLimit = await this.provider.estimateGas({
-        from: this.entryPoint.address,
-        to: this.getAccountAddress(),
-        data: callData
-      })
-      // if wallet is not deployed we need to multiply estimated limit to 3 times to get accurate callGasLimit
+      try {
+        callGasLimit = await this.provider.estimateGas({
+          from: this.entryPoint.address,
+          to: this.getAccountAddress(),
+          data: callData
+        })
+      } catch (error) {
+        Logger.error(' Call Gas Limit Estimation Failed ')
+        if (detailsForUserOp && detailsForUserOp.target.length === 1 &&
```

Comment:
what is this piece of code doing? when it will fall into catch block and only based on this check custom gas price is applied..

Reply:
in zkevm polygon case, it falls to catch block, we try to give hardcoded value but only in case its not just wallet deployment transaction. Because for wallet deployment we already have a hardcode number of 500k

Reply:
why can't it estimate gas for zkevm?

Reply:
also what is the point of this?

if (detailsForUserOp && detailsForUserOp.target.length === 1 && detailsForUserOp.target[0] !== AddressZero)

how does it ensure it's just wallet deployment transaction? if to address is not zero it's wallet deployment, how?

Reply:
> why can't it estimate gas for zkevm?

it just throw error saying. Gas estimation failed, unpredictable gas limit

Reply:
> also what is the point of this?
> 
> if (detailsForUserOp && detailsForUserOp.target.length === 1 && detailsForUserOp.target[0] !== AddressZero)
> 
> how does it ensure it's just wallet deployment transaction? if to address is not zero it's wallet deployment, how?

Han let say i am just doing wallet deployment using AA then detailsForUserOp.target length would be one and to is going to be address zero other then that we will just use hardcoded gaslimit

Reply:
single transaction and to address non-zero -> how does it signify wallet deployment?!

Reply:
I would still buy it if was check for 0 address

Reply:
> single transaction and to address non-zero -> how does it signify wallet deployment?!

Han to address can be zero for other transaction i will just throw error if we are not able to estimate gas. 

Will post issue in zkevm channel regarding gas estimation